src/report/metrics.js

```diff
@@ -80,6 +80,11 @@ function transformMetricsToTable(
   lComponents = lComponents.concat(
     hideModules ? [] : modules.filter(metricsAreCalculable).map(metricifyModule)
   );
+    
+  const collapsePatternRegex = new RegExp(collapsePattern, 'g');
+  lComponents = lComponents
+    .filter(({ name }) => name.match(collapsePatternRegex))
```

Comment:
Although it takes a collapsePattern it performs a filter and not a collapse/ aggregation (which would have to consolidate module counts, Ca/ Ce and re-calculate instability - which currently not even the module [consolidation](https://github.com/sverweij/dependency-cruiser/blob/develop/src/graph-utl/consolidate-to-pattern.js) takes into account). 

It seems the intent of this code is more like what --include-only (/ includeOnly) is for. For command line use with `depcruise-fmt`/ the reporter API (as indicated in the PR description) adding a folder filter in [main/report-wrap.js](https://github.com/sverweij/dependency-cruiser/blob/develop/src/main/report-wrap.js) then might be a better fit. As a bonus other reporters that take folders into account will then benefit as well.